### PR TITLE
fix(examples): return focus to trigger when drawer modal is closed

### PR DESCRIPTION
**Pull request by Francois-Esquire**
## Description

Adds the `restoreFocus` and `focusOnMount` props to the default `DrawerModal` example to return focus to the trigger element.

## Checklist

- [ ] :ok_hand: ~website updates are Garden Designer approved (add the designer as a reviewer)~
- [ ] :black_nib: ~copy updates are approved (add the content strategist as a reviewer)~
- [ ] :link: ~considered opportunities for adding cross-reference URLs (grep for keywords)~
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/examples/drawer/DrawerDefault.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const Example = () => {
     <Row>
       <Col textAlign="center">
         <Button onClick={open}>Open Drawer</Button>
-        <DrawerModal isOpen={isOpen} onClose={close}>
+        <DrawerModal isOpen={isOpen} onClose={close} restoreFocus focusOnMount>
```
Comment (jzempel) [inline src/examples/drawer/DrawerDefault.tsx:23]: This isn't working for me. When I pull the example into CodeSandbox, only `restoreFocus` is needed to restore keyboard focus back to the trigger button (just like the `Modal` example).

**Review by jzempel — CHANGES_REQUESTED** [review]

File: src/examples/drawer/DrawerDefault.tsx (hunk 1 above)
Comment (Francois-Esquire) [reply src/examples/drawer/DrawerDefault.tsx:23]: Thats correct, I am in the process of investigating as to why this isn't working as expected and it looks like focus is reverting to `#gatsby-focus-wrapper` for some odd reason. I will continue to look more into this.

Comment (jzempel) [reply src/examples/drawer/DrawerDefault.tsx:23]: Did you try in production mode? I seem to remember we might've seen some odd Gatsby stuff with this in development.

Comment (Francois-Esquire) [reply src/examples/drawer/DrawerDefault.tsx]: It didn't seem to work in development mode as well...

**Review by jzempel — APPROVED** [review]

**Review by geotrev — APPROVED** [review]
LGTM

**Review by jzempel — CHANGES_REQUESTED** [review]
On second thought, I think there's more to do here with unmounting + animation. When I matched the default `<Modal>` example, the `restoreFocus` works as intended.

**Comment by Francois-Esquire** [comment]
> On second thought, I think there's more to do here with unmounting + animation. When I matched the default `<Modal>` example, the `restoreFocus` works as intended.

Yes, mounting/unmounting the component does work however the animation is completely void when doing so.

**Comment by geotrev** [comment]
Latest deployment from my PR includes the drawer change: https://63ea6942dbe9e40052c2e721--zendeskgarden.netlify.app/

Focus is working correctly now. 👍🏻

**Pull request closed without merging**